### DISCUSSION TITLE
Making Github repos "first class citizens"

### DISCUSSION THREAD

**Discussion by JoshPaulie** [pypa/pipx#1335] [Ideas]
# Proposal
I'm proposing **pipx** accept specs with the following format: `username/repo`
Which would simply be expanded to `git+https://github.com/username/repo.git` under the hood.

This would provide a simpler way for Python developers to distribute their projects, without the need to take up some of PyPI's namespace.

## Aesthetic
I'd argue the former is a lot neater to leave on your README.md than the latter

```
pipx install username/repo
pipx install git+https://github.com/username/repo.git
```

## Precedent
Other communities, like NeoVim's plugin community, use this pattern to install plugins. Although it might seem a bit cumbersome to not only remember the app name but also the author's name, you quickly adjust.

## Other repository hosts
Would not be supported.

**Comment by chrysle** [comment]
My opinion on that mostly coincides with that of @gaborbernat in https://github.com/pypa/pipx/issues/812#issuecomment-1907182045.

**Reply by JoshPaulie** [reply]
Seems unrelated. No need to integrate the GitHub api. It'd be a matter of checking the passed spec for a pattern matching "username/repo", and if so, interpolate.

**Reply by chrysle** [reply]
>  and if so, interpolate.

However that's GitHub-specific and would most probably call in people who make use of other frequently used Git hosting services like GitLab or Codeberg and want the same feature. I don't think GH should be preferred to other services.

**Reply by Gitznik** [reply]
I have to agree with @chrysle here.

Technically we could do what `go get` does (supporting install via `github/user/repo` among other git hosting providers), but that's a non-trivial task and IMO not necessary for the fairly minor gain in readability.

**Reply by JoshPaulie** [reply]
I guess that's that! I still believe it would greatly simplify distributing Python apps.

Thanks for the discussion! Big proponent of pipx, such a great addition to the ecosystem. 😄
